btcutil/psbt/partial_input.go

```diff
@@ -13,16 +13,22 @@ import (
 // PInput is a struct encapsulating all the data that can be attached to any
 // specific input of the PSBT.
 type PInput struct {
-	NonWitnessUtxo     *wire.MsgTx
-	WitnessUtxo        *wire.TxOut
-	PartialSigs        []*PartialSig
-	SighashType        txscript.SigHashType
-	RedeemScript       []byte
-	WitnessScript      []byte
-	Bip32Derivation    []*Bip32Derivation
-	FinalScriptSig     []byte
-	FinalScriptWitness []byte
-	Unknowns           []*Unknown
+	NonWitnessUtxo         *wire.MsgTx
+	WitnessUtxo            *wire.TxOut
+	PartialSigs            []*PartialSig
+	SighashType            txscript.SigHashType
+	RedeemScript           []byte
+	WitnessScript          []byte
```

Comment:
Non-blocking, but we may want to consider using struct embedding to make the set of fields more preamble at a glance. Since the structs would be embed, there wouldn't be any additional layers of indirection required for callers.

Reply:
You mean to create a `TaprootPInput` that can be embedded to avoid having all those fields with the `Taproot...` prefix? We could do that, yes. Though we'd get a collision with the `Bip32Derivation` derivation name, not sure if that would lead to more mistakes or confusion?